internal/lexer/token.go

```diff
@@ -0,0 +1,51 @@
+package lexer
+
+type TokenType int
+
+const (
+	EOF TokenType = iota
+	IDENTITY
+	NUMBER
+	STRING
+
+	PATH     // .
+	PIPE     // |
+	LBRACKET // [
+	RBRACKET // ]
+
+	beginOperatorSection
+	ASSIGN   // =
+	EQUALITY // ==
+	MERGE    // +
+	endOperatorSection
+)
+
+var tokens = []string{
+	EOF:      "EOF",
+	IDENTITY: "IDENTITY",
+	NUMBER:   "NUMBER",
+	STRING:   "STRING",
+
+	PATH:     "PATH",
+	PIPE:     "PIPE",
+	LBRACKET: "LBRACKET",
```

Comment:
I don't think it should be part of the tokens. It is relevant to capturers that are referenced via another capturer only. I believe it will be easier to tokenize the whole referenced capturer. It will be the responsibility of the resolver to understand how to access the specified capturer path.

Reply:
Agree brackets has to be understood by the lexer and just returning it with path, at the end brackets is just syntatic suggar.